src/apps/d2dMultihop/MultihopD2D.cc

```diff
@@ -294,28 +319,50 @@ void MultihopD2D::handleTrickleTimer(cMessage* msg)
 
 void MultihopD2D::relayPacket(cMessage* msg)
 {
-    MultihopD2DPacket* pkt = check_and_cast<MultihopD2DPacket*>(msg);
+
+    Packet* pPacket = check_and_cast<Packet*>(msg);
+    auto src = pPacket->popAtFront<MultihopD2DPacket>();
+
+    // create a relay packet using the data of the source packet
+    auto data = pPacket->popAtFront<ByteCountChunk>();
```

Comment:
This popAtFront will not work when working with serialized packets, because it implicitly assumes the the representation is actualy a ByteCountChunk, which may or may not what you want to do here. If you want to process all remaining data up to the back pointer of the packet, then you should use peekData(), etc.

Reply:
Was changed to peekData().